bin/publish.sh

```diff
@@ -62,51 +88,36 @@ else
   actions="$*"
 fi
 
-export GOOS=linux
-export GOARCH=amd64
-
-function print_info() {
-  for info in "$@"; do
-    case "$info" in
-    image)
-      echo -e "image:\n  image_url: ${image_url}"
-      ;;
-    *)
-      echo "unknown info: ${info}" >&2
-    esac
-  done
-}
-
-if [[ "$V" -gt 0 ]]; then
-  set -x
-fi
 for action in $actions; do
   case "$action" in
   build)
-    go build -o manager.exe
+    echo "go build submodules ${MOD}"
+    CGO_ENABLED="${CGO_ENABLED}" GOOS="${TARGET_GOOS}" GOARCH=${TARGET_GOARCH}  go build -o manager.exe
     ;;
   image)
-    docker build -t "${image_url}" .
-    ;;
-  print)  # should be the only action
-    # rest param will be consider as info to print, like: *.sh print image
-    set +x
-    shift
-    print_info "$@"
-    break
-    ;;
-  push)
-    docker push "${image_url}"
+    echo "build docker image: ${image}" >&2
+    if [[ "$TARGET_GOOS" == "linux" && "$TARGET_GOARCH" == "amd64" && ("$(uname -p)" == "x86_64" || "$(uname -p)" == "i386") ]]; then
```

Comment:
I found that this condition-branch is meaningless, we can remove it and use `docker buildx build` line directly